src/charm.py

```diff
@@ -202,27 +202,43 @@ def _configure_certificate_requirers(self):
         if not gocert_csrs_table:
             logger.warning("couldn't distribute certificates: couldn't get table from GoCert")
             return
+
         for request in databag_csrs:
+            # Find all rows in the table that match the request's CSR
             matching_rows = list(
                 filter(
                     lambda x: x.csr == str(request.certificate_signing_request),
                     gocert_csrs_table.rows,
                 )
             )
+            # If there are no matches, upload CSR to GoCert
             if len(matching_rows) < 1:
                 self.client.post_csr(str(request.certificate_signing_request), login_details.token)
                 continue
-            request_row = matching_rows[0]
-            provided_certificates = self.tls.get_issued_certificates(request.relation_id)
-            if request_row.certificate_chain == "rejected" or request_row.certificate_chain == "":
-                # TODO: tls lib v4 lacks the ability to revoke the certificate of a specific request
+            matching_row = matching_rows[0]
+            matching_provided_certificates = list(
+                filter(
+                    lambda x: str(x.certificate_signing_request) == matching_row.csr,
+                    self.tls.get_issued_certificates(request.relation_id),
+                )
+            )
+            if (
+                matching_row.certificate_chain == "rejected"
+                or matching_row.certificate_chain == ""
+            ):
+                # TODO: tls lib v4 lacks the ability to remove the certificate of a specific request
```

Comment:
I see the TODO has changed here. Is this something that we would require here? If so, I would propose creating an issue in the lib and linking it here.

Reply:
done